app/code/Magento/CatalogGraphQl/Model/MediaGalleryTypeResolver.php

```diff
@@ -23,11 +23,10 @@ class MediaGalleryTypeResolver implements TypeResolverInterface
     public function resolveType(array $data) : string
     {
         // resolve type based on the data
-        if (isset($data['media_type']) && $data['media_type'] == 'image') {
-            return 'ProductImage';
-        }
-        if (isset($data['media_type']) && $data['media_type'] == 'external-video') {
+        if (!empty($data['video_content'])) {
```

Comment:
How it will work with code on mainline 2.4-develop (without our repos)?

If it will work in an expected way we can leave it here and later merge to 2.4-develop, if not - we need to solve this in GQL resolver in "storefront" repo - just add "media_type" to be able determinate type correctly

Reply:
yes, this works w/o storefront repos with 2.4-develop

To resolve in "storefront" repo, you mean to create a preference for MediaGalleryTypeResolver or resolve in DataMapper and save "media_type" into ES?